src/components/common/StaffComments.vue

```diff
@@ -0,0 +1,252 @@
+  private get numComments (): string {
+    const numComments = this.comments.length
+    return (numComments === 1 ? '1 Comment' : `${numComments} Comments`)
+  }
+
+  /** Array of validations rules for the textarea. */
+  private get rules (): Array<Function> {
+    // exclude whitespace in minimum length check
+    // include whitespace in maximum length check
+    return [
+      val => (val && val.trim().length > 0) || 'Enter a comment.',
+      val => (val && val.length <= this.maxLength) || 'Maximum characters reached.'
+    ]
+  }
+
+  /** Called when the component is mounted. */
+  private async mounted (): Promise<void> {
+    await this.fetchStaffComments()
+  }
+
+  /** Fetches the staff comments from the API. */
+  private async fetchStaffComments (): Promise<void> {
+    const url = `businesses/${this.businessId}/comments`
+    this.comments = await this.axios.get(url)
+      .then(resp => this.flattenAndSortComments(resp?.data?.comments))
+      .catch(() => [])
+  }
+
+  /** Saves the new comment to the API. */
+  private async save (): Promise<void> {
+    // don't save if invalid
+    if (!this.$refs.textarea.validate()) return
+
+    // prevent double saving
+    if (this.isSaving) return
+    this.isSaving = true
+
+    const url = `businesses/${this.businessId}/comments`
+    const data = {
+      comment: {
+        businessId: this.businessId,
+        comment: this.comment
+      }
+    }
+
+    let success = false
+    await this.axios.post(url, data).then(res => {
+      success = true
+    }).catch(error => {
+      // eslint-disable-next-line no-console
+      console.log('save() error =', error)
+      alert('Could not save your comment. Please try again or cancel.')
+    })
+
+    this.isSaving = false
+    if (success) {
+      // clear the data and reload the staff comments
+      this.$refs.textarea.reset()
+      await this.fetchStaffComments()
+    }
+  }
+
+  /** Closes the menu (dialog). */
+  private close (): void {
+    // clear any errors; leave the data
+    this.$refs.textarea.resetValidation()
+    this.showComments = false
+  }
+}
+</script>
+
+<style lang="scss" scoped>
+@import '@/assets/styles/theme.scss';
+
+#staff-comment-container {
+  width: 33rem;
+  height: 36rem;
+  overflow: hidden;
+}
+
+.v-card__title {
+  .v-icon {
+    margin-top: 1px;
+  }
+
+  span {
+    color: $primary-blue;
+    font-size: 0.75rem;
+    letter-spacing: normal;
+    margin-left: 5px;
+  }
+}
+
+::v-deep .v-textarea textarea {
+  font-size: 0.875rem !important;
+  color: $gray7 !important;
+}
+
+// reduce overall textarea height when there are no error messages
+::v-deep .v-textarea:not(.error--text) {
+  margin-bottom: -24px;
+}
+
+// reduce overall textarea height when there are errors messages
+::v-deep .v-textarea.error--text {
+  margin-bottom: -12px;
+}
+
+// shrink input area to make space for error messages
+::v-deep .v-textarea.error--text textarea {
+  height: calc(140px - 12px) !important;
```

Comment:
Unfortunately this is hard-coded, but the sizes are static anyway.

The reason for this CSS rule is so that an error message doesn't push anything below it down the page. The reason for that is that the click event on v-btns doesn't work reliably when the button moves while you're clicking on it!!! This took me a while to figure out.